### ISSUE TITLE
Better handling of enums

### ISSUE THREAD

**Issue by RatTac** [mrjono1/joi-to-typescript#147]
Hey,
overall a very nice package!

I noticed an issue when using enums:

```typescript
import Joi from "joi";

export const roleSchema = Joi.string()
	.valid("Admin", "User")
	.label("Role");

export const userSchema = Joi.object({
	firstName: Joi.string().required(),
	lastName: Joi.string().required(),
	role: roleSchema.required(),
}).label("User");
```
This will generate the following output:
```typescript
export type Role = 'Admin' | 'User';

export interface User {
	firstName: string;
	lastName: string;
	role: Role | 'Admin' | 'User';
}
```
However `role: Role` would be enough and and more readable ;)
Regards

**Comment by mrjono1** [comment]
I have a fix for this! I am currently working on it

**Comment by mrjono1** [comment]
https://github.com/mrjono1/joi-to-typescript/pull/320 after this PR is merged swap `.allow()` to `.valid()` or `.allow().only()` then the out put will be correct

**Comment by mrjono1** [comment]
```typescript
import Joi from "joi";

export const roleSchema = Joi.string()
	.valid("Admin", "User")
	.meta({ className: "Role" });

export const userSchema = Joi.object({
	firstName: Joi.string().required(),
	lastName: Joi.string().required(),
	role: roleSchema.required(),
}).meta({ className: "User" });
```
```typescript
export type Role = 'Admin' | 'User';

export interface User {
	firstName: string;
	lastName: string;
	role: Role;
}
```

**Comment by mrjono1** [comment]
released under 4.3.0
